eth/state_accessor.go

```diff
@@ -84,22 +65,27 @@ func (eth *Ethereum) StateAtBlock(ctx context.Context, block *types.Block, reexe
 		if preferDisk {
 			// Create an ephemeral trie.Database for isolating the live one. Otherwise
 			// the internal junks created by tracing will be persisted into the disk.
-			database = state.NewDatabaseWithConfig(eth.chainDb, &trie.Config{Cache: 16})
+			// TODO(rjl493456442), clean cache is disabled to prevent memory leak,
+			// please re-enable it for better performance.
+			database = state.NewDatabaseWithConfig(eth.chainDb, trie.HashDefaults)
```

Comment:
Why leave the comment vs doing a Close on the database to free the cache?

Reply:
It's a bit complicated, I plan to do it in the following PR.

We have the reference/dereference operation which are done in release function,
but the db is shared by all the states created on top. We can't simply close it in
release function.

It's definitely in my radar, just need to figure out how to solve it elegantly.